contracts/extensions/version-control/ConditionalImplementControl.sol

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { ERC1967Upgrade } from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+import { IConditionalImplementControl } from "../../interfaces/version-control/IConditionalImplementControl.sol";
+import { ErrorHandler } from "../../libraries/ErrorHandler.sol";
+import { AddressArrayUtils } from "../../libraries/AddressArrayUtils.sol";
+import { ErrOnlySelfCall } from "../../utils/CommonErrors.sol";
+
+/**
+ * @title ConditionalImplementControl
+ * @dev A contract that allows conditional version control of contract implementations.
+ */
+abstract contract ConditionalImplementControl is IConditionalImplementControl, ERC1967Upgrade {
```

Comment:
If "ConditionalImplement`ation`Control" is a too-long name, could we name it "ImplementationSwitcher"?

Reply:
I want to keep `Conditional`.